doc/introduction/interfaces/numpy.rst

```diff
@@ -110,14 +110,66 @@ with respect to both QNode parameters ``phi`` and ``theta``:
 
     phi = np.array([0.5, 0.1])
     theta = 0.2
-    dcircuit = qml.grad(circuit3, argnum=[0, 1])
+    dcircuit = qml.grad(circuit3)
 
 Evaluating this gradient function at specific parameter values:
 
 >>> dcircuit(phi, theta)
 (array([ -4.79425539e-01,   1.11022302e-16]), array(0.0))
 
 
+Non-differentiable arguments
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Note that **all positional arguments provided to the QNode are assumed to be differentiable
+by default**:
+
+>>> from pennylane import numpy as np
+>>> np.array([0.1, 0.2])
+tensor([0.1, 0.2], requires_grad=True)
+
+If you would like to provide explicit non-differentiable arguments to the
+QNode or gradient function, make sure to use a NumPy array that specifies
+``requires_grad=False``:
+
+>>> np.array([0.1, 0.2], requires_grad=False)
+tensor([0.1, 0.2], requires_grad=False)
+
+The ``requires_grad`` argument can be passed to any NumPy function provided by
+PennyLane.
+
+For example, consider the following QNode that accepts one trainable argument ``weights``,
+and two non-trainable arguments ``data`` and ``wires``:
+
+.. code-block:: python
+
+    dev = qml.device('default.qubit', wires=5)
+
+    @qml.qnode(dev)
+    def circuit(weights, data, wires):
+        qml.templates.AmplitudeEmbedding(data, wires=wires, normalize=True)
+        qml.RX(weights[0], wires=wires[0])
+        qml.RY(weights[1], wires=wires[1])
+        qml.RZ(weights[2], wires=wires[2])
+        qml.CNOT(wires=[wires[0], wires[1]])
+        qml.CNOT(wires=[wires[0], wires[2]])
+        return qml.expval(qml.PauliZ(wires[0]))
+
+We must specify that ``data`` and ``wires`` are NumPy arrays with ``requires_grad=False``:
+
+>>> weights = np.array([0.1, 0.2, 0.3])
+>>> data = np.random.random([2**3], requires_grad=False)
+>>> wires = np.array([2, 0, 1], requires_grad=False)
+>>> circuit(weights, data, wires)
+0.16935626052294817
+
```

Comment:
Shall we add something along the lines of 

If you do not want to specify the `requires_grad` flag, simply create a QNode that accepts the non-differentiable arguments as keyword arguments:

[Example]

This will also work if you create arrays after importing vanilla numpy with ``import numpy as np``.

Reply:
Maybe fits better at the end...